DMCompiler/DM/Expressions/Builtins.cs

```diff
@@ -77,6 +77,21 @@ class NewPath : DMExpression {
         }
     }
 
+    class NewMultidimensionalList : DMExpression {
+        DreamPath TargetPath;
+        ArgumentList Arguments;
+
+        public NewMultidimensionalList(Location location, DreamPath targetPath, ArgumentList arguments) : base(location) {
+            TargetPath = targetPath;
+            Arguments = arguments;
+        }
+
+        public override void EmitPushValue(DMObject dmObject, DMProc proc) {
+            Arguments.EmitPushArguments(dmObject, proc);
```

Comment:
There's no need to turn the sizes into arguments. You could keep a list of all the expressions, push them onto the stack, and give the `CreateMultidimensionalList` opcode an immediate value describing how many dimensions (a.k.a expressions pushed onto the stack) there are.